src/main/java/fi/fmi/avi/archiver/config/MessagePopulatorConfig.java

```diff
@@ -25,6 +26,7 @@
     private final List<PopulatorInstanceSpec> executionChain;
 
     MessagePopulatorConfig(final List<PopulatorInstanceSpec> executionChain) {
+        checkState(!executionChain.isEmpty(), "Message populator execution chain not configured");
         this.executionChain = requireNonNull(executionChain, "executionChain");
```

Comment:
To have a message in possible `NullPointerException`, the `requireNonNull()` invocation should be moved from next line to previous. Or simply move this line after next line.

```suggestion
        this.executionChain = requireNonNull(executionChain, "executionChain");
        checkState(!executionChain.isEmpty(), "Message populator execution chain not configured");
```

Reply:
Oops. Fixed.